raiden/transfer/channel.py

```diff
@@ -1221,11 +1341,13 @@ def create_unlock(
     locksroot = merkleroot(merkletree)
 
     token_address = channel_state.token_address
-    nonce = get_next_nonce(our_state)
     recipient = channel_state.partner_state.address
     # the lock is still registered
     locked_amount = TokenAmount(get_amount_locked(our_state) - lock.amount)
 
+    nonce = get_next_nonce(our_state)
+    channel_state.our_state.nonce = nonce
```

Comment:
`create` functions should not change the state of the objects, please move this to the `send_`

Reply:
This was done because in some of our tests, we call create_sendlockedtransfer to create a transfer by the partner (to simulate receiving a transfer from our partner)


in this case, the nonce of the partner will be increased in the create_send* functions which will fail the tests.